cpg_workflows/jobs/bam_to_cram.py

```diff
@@ -55,3 +55,46 @@ def bam_to_cram(
     j.command(command(cmd, monitor_space=True))
 
     return j, j.sorted_cram
+
+
+def cram_to_bam(
+    b: Batch,
+    input_cram: ResourceGroup,
+    extra_label: str | None = None,
+    job_attrs: dict | None = None,
+    requested_nthreads: int | None = None,
+) -> tuple[Job, ResourceGroup]:
+    """
+    Convert a CRAM file to a BAM file.
+    """
+    
+    assert isinstance(input_cram, ResourceGroup)
+
+    job_name = 'cram_to_bam'
+    if extra_label:
+        job_name += f' {extra_label}'
+
+    convert_tool = 'samtools_view_cram_to_bam'
+    j_attrs = (job_attrs or {}) | dict(label=job_name, tool=convert_tool)
+    j = b.new_job(name=job_name, attributes=j_attrs)
+    j.image(image_path('samtools'))
+
+    # Set resource requirements
+    nthreads = requested_nthreads or 8
+    res = STANDARD.set_resources(
+        j,
+        ncpu=nthreads,
+        storage_gb=50,  # TODO: make configurable
+    )
+
+    j.declare_resource_group(
+        sorted_bam={
+            'bam': '{root}.bam',
+            'bam.bai': '{root}.bam.bai',
+        }
+    )
+
+    cmd = f'samtools view -@ {res.get_nthreads() - 1} -b {input_cram.cram} | tee {j.sorted_bam["bam"]} | samtools index -@ {res.get_nthreads() - 1} - {j.sorted_bam["bam.bai"]}'
```

Comment:
this puts the resulting BAM file in _very_ temporary this-batch-only storage (i.e. if a job in the batch fails then this whole inflation needs to be repeated). Would it be worth copying this into the project temp (config.storage.default.temp e.g. `cpg-<whatever>-main-tmp`) and then putting in a loop that will either use or regenerate, depending on whether it already exists?

Reply:
Thanks for pointing that out. Turns out, I had done that, just in one of the downstream branches. I'm updating the code now so it's in this branch.